### PR TITLE
Business manager dashboard

### PR DESCRIPTION
**Pull request by boveus**
Sorry for the big commit, guys.  There was a lot going into this user story.  This should give us a good baseline for the multi-tenancy auth, though.

I added the business manager dashboard.  This was kind of tough due implementing the multi-tenancy and I am not sure if this is the exact way to do it, but I followed the lesson plan more or less.

I didn't implement any styling for the business manager dashboard and the layout is super simple, but it has some of the initial functionality we will need.

We might be able to refactor how I am using the factories in the tests, since I didn't really use factorygirl to its fullest extent.  It ended up taking more time to make factorygirl work than was worth it, so I handrolled my tests a bit.

Let me know what y'all think

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -3,9 +3,16 @@ class User < ApplicationRecord
   validates_presence_of :name, :email, :address, :password
   validates_uniqueness_of :name, :email
   has_many :orders
-  
-  enum role: [:default, :admin]
-  
+  has_many :user_roles
+  has_many :roles, through: :user_roles
+  belongs_to :vendor, optional: true
```
Comment (erose357) [inline app/models/user.rb]: where'd you find the `optional: true`?, that's pretty cool

File: app/models/vendor.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 class Vendor < ApplicationRecord
+  has_many :users
```
Comment (erose357) [inline app/models/vendor.rb]: This confused me at first, but now it makes sense.  Is this so you can identify a 'business_admin' by  the relationship to the vendor?  cool

File: db/schema.rb [hunk 3]
```diff
@@ -54,12 +54,27 @@
...
     t.datetime "updated_at", null: false
   end
 
+  create_table "user_roles", force: :cascade do |t|
+    t.bigint "user_id"
+    t.bigint "role_id"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["role_id"], name: "index_user_roles_on_role_id"
+    t.index ["user_id"], name: "index_user_roles_on_user_id"
```
Comment (samuelssnider) [inline db/schema.rb]: Did this come from a rails g?

File: spec/features/business_manager/business_manager_logs_in_and_sees_dashboard_spec.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    fill_in "user[password]", with: "Password"
+    click_on("Log in")
+
+    save_and_open_page
+    expect(page).to have_link(user_vendor.name)
+    expect(page).to_not have_link(Vendor.second.name)
+    expect(page).to_not have_link(Vendor.third.name)
+    expect(page).to_not have_link(Vendor.fourth.name)
+    expect(page).to_not have_link(Vendor.fifth.name)
+    end
```
Comment (samuelssnider) [inline spec/features/business_manager/business_manager_logs_in_and_sees_dashboard_spec.rb]: Test looks good!

**Pull request merged**